src/modules/hip/kernel/snow.cpp

```diff
@@ -102,58 +130,65 @@ extern "C" __global__ void snow_pln( unsigned char *output,
                 {
                     int id = i + j * width;
-                    for(int k = 0;k < channel ; k++)
-                        output[pixIdx + rand_id + id + ( k * channelSize)] = snow_mat[i][j];
+                    for(int k = 0; k < channel; k++)
+                    {
+                        output[pixIdx + rand_id + id + (k * channelSize)] = snow_mat[i][j];
+                    }
                 }
             }
         }
-        
     }
 }
 
-
-
-extern "C" __global__  void snow_batch( unsigned char* input,
-                                   unsigned char* output,
-                                   const float *snowPercentage,
-                                   const unsigned int *height,
-                                   const unsigned int *width,
-                                   const unsigned int *max_width,
-                                   const unsigned long *batch_index,
-                                   const unsigned int channel,
-                                   const unsigned int *inc, // use width * height for pln and 1 for pkd
-                                   const int plnpkdindex // use 1 pln 3 for pkd
-                                    )
+extern "C" __global__ void snow_batch(unsigned char *input,
+                                      unsigned char *output,
+                                      const float *snowPercentage,
+                                      const unsigned int *height,
+                                      const unsigned int *width,
+                                      const unsigned int *max_width,
+                                      const unsigned long long *batch_index,
+                                      const unsigned int channel,
+                                      const unsigned int *inc, // use width * height for pln and 1 for pkd
+                                      const int plnpkdindex) // use 1 pln 3 for pkd
 {
     int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
     int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
     int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
     float rainProbTemp = snowPercentage[id_z];
     unsigned int snowHeightTemp = 5;
     unsigned int snowWidthTemp = 5;
-    int indextmp=0;
+    int indextmp = 0;
     long pixIdx = 0;
     int rand;
-    int snow_mat[5][5] = {{0,50,75,50,0}, {40,80,120,80,40}, {75,120,255,120,75}, {40,80,120,80,40}, {0,50,75,50,0}};
+
     if(id_x < width[id_z] && id_y < height[id_z])
     {
-        pixIdx = batch_index[id_z] + (id_x  + id_y * max_width[id_z] ) * plnpkdindex ;
+        pixIdx = batch_index[id_z] + (id_x + id_y * max_width[id_z]) * plnpkdindex;
         if((id_y >= 0 ) && (id_y < height[id_z]) && (id_x >= 0) && (id_x < width[id_z]))
-        {   
+        {
             float pixelDistance = 1.0;
-            pixelDistance /=  (rainProbTemp / 100);
+            pixelDistance /= (rainProbTemp / 100);
```

Comment:
avoid this division in the low level kernel. Please pass float parameter from high level

Reply:
We'll do the kernel level optimizations subsequent PRs